### PR TITLE
.github: Consolidate CI workflows into one

### PR DESCRIPTION
**Pull request by dave-tucker**
This gives a better view of the CI pipeline in Github. Gives us control over what runs and when.
And finally, yields a single "build-workflow-passed" status that can be used for merge automation later.

### CODE REVIEW

**Review by tamird — COMMENTED** [review]
+1 to having all the workflows in one file, thanks for doing this!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
+name: aya-ci
+
+on:
+  push:
+    branches:
+      - main
+      - ci
```
[the remark this answers is not in the document]
Comment (dave-tucker) [reply .github/workflows/ci.yml:7]: We did once upon a time but it's not strictly necessary

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          cargo hack miri test --all-targets --feature-powerset \
+            --exclude aya-bpf \
+            --exclude aya-bpf-bindings \
+            --exclude aya-log-ebpf \
+            --exclude integration-ebpf \
+            --exclude integration-test \
+            --workspace
+
+  build-test-aya:
+    needs: ["lint"]
```
[the remark this answers is not in the document]
Comment (dave-tucker) [reply .github/workflows/ci.yml:51]: Yep

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            .tmp/test_rsa.pub
+          # FIXME: we should invalidate the cache on new bpf-linker releases.
+          # For now we must manually delete the cache when we release a new
+          # bpf-linker version.
+          key: tmp-files-${{ hashFiles('test/run.sh') }}
+
+      - name: Run integration tests
+        run: test/run.sh
+
+  build-workflow-complete:
```
[the remark this answers is not in the document]
Comment (dave-tucker) [reply .github/workflows/ci.yml:163]: Not with Mergify no. It's a [known issue](https://docs.mergify.com/conditions/#validating-all-status-checks) and this is a workaround I've used in other projects.

[the remark this answers is not in the document]
Comment (tamird) [reply .github/workflows/ci.yml:163]: Got it. Can you leave a comment?

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tamird) [reply .github/workflows/ci.yml:7]: Should we remove it? What's it for?

[the remark this answers is not in the document]
Comment (dave-tucker) [reply .github/workflows/ci.yml]: Done

**Review by astoycos — APPROVED** [review]
/lgtm 

Thanks!

**Pull request merged**